halley-core/src/main/kotlin/com/infinum/halley/core/loader/HalRelationshipLoader.kt

```diff
@@ -20,11 +21,16 @@ internal class HalRelationshipLoader : RelationshipLoader {
 
             val requestUrl = appendParameters(url, request.name, request.options)
 
-            CallFactoryCache.load().newCall(
+            val call = CallFactoryCache.load().newCall(
                 Request.Builder()
                     .url(requestUrl)
                     .build()
-            ).enqueue(callback)
+            )
+            try {
+                call.enqueue(callback)
+            } catch (ex: InterruptedException) {
+                call.cancel()
+            }
```

Comment:
It's not good; we need to try-catch the `await` part (as well). Since await is the blocking function, enqueue is async, so it completes immediately.

Reply:
Ahh right, my bad, I’ll fix it, thanks!